util/generate-ami-list.py

```diff
@@ -189,19 +228,23 @@ def update_cfn_template(cfn_template_file, amis_to_update):
         # object_pairs_hook=OrderedDict allows to preserve input order
         cfn_data = json.load(cfn_file, object_pairs_hook=OrderedDict)
     # update id for new amis without removing regions that are not in the amis_to_update dict
-    current_amis = cfn_data.get("Mappings").get("AWSRegionOS2AMI")
-    current_amis.update(amis_to_update)
-    # enforce alphabetical regions order
-    ordered_amis = OrderedDict(sorted(current_amis.items()))
-    cfn_data.get("Mappings")["AWSRegionOS2AMI"] = ordered_amis
+    current_amis = get_initialized_mappings_dicts()
+    for mapping_name, regions in current_amis.items():
+        current_amis_for_mapping = cfn_data.get("Mappings").get(mapping_name, {})
+        for region, os_to_amis in current_amis_for_mapping.items():
```

Comment:
Could we just do `current_amis_for_mapping.update(amis_to_update.get(mapping_name, {}))` here?

Reply:
The reason it's done this way is to avoid the following:
```
>>> d1 = {'TopKey': {'SubkeyOne': 'SubvalueOne'}}
>>> d2 = {'TopKey': {'SubkeyTwo': 'SubvalueTwo'}}
>>> d1.update(d2)
>>> d1
{'TopKey': {'SubkeyTwo': 'SubvalueTwo'}}
```

In most cases I'd say this should be handled by a generic dict-merging function like the one we make heavy use of in the unit tests to easily express a config dict that differs from the defaults. But for one the structure of the mappings is known, and for another I like how spelling out the logic makes it more apparent how the function's affecting the structure that will eventually become the output of the script.